web/crux/src/app/version/version.service.ts

```diff
@@ -71,7 +78,15 @@ export default class VersionService {
     return versions > 0
   }
 
-  async getVersionsByProductId(productId: string, user: Identity): Promise<VersionDto[]> {
+  async getVersionsByProductId(productId: string, user: Identity, query?: VersionListDto): Promise<VersionDto[]> {
+    const filter = {
```

Comment:
This should has a type in prisma i believe. Please use that like: `Prisma.typename`